googletest/include/gtest/internal/gtest-port.h

```diff
@@ -603,6 +603,13 @@ struct _RTL_CRITICAL_SECTION;
     || GTEST_OS_QNX || GTEST_OS_FREEBSD || GTEST_OS_NACL)
 #endif  // GTEST_HAS_PTHREAD
 
+// On MinGW, we can have both GTEST_OS_WINDOWS and GTEST_HAS_PTHREAD
+// defined, but we don't want to use MinGW's pthreads implementation, which
+// has conformance problems with some versions of the POSIX standard.
+#if GTEST_HAS_PTHREAD && GTEST_OS_WINDOWS_MINGW
+#undef GTEST_HAS_PTHREAD
```

Comment:
This is weird. We're using GTEST_HAS_PTHREAD as a check to undef GTEST_HAS_PTHREAD.
Let's just change where the GTEST_HAS_PTHREAD was first defined.